tools/schema_loader.cc

```diff
@@ -318,6 +323,217 @@ std::vector<schema_ptr> do_load_schemas(std::string_view schema_str) {
     return schemas;
 }
 
+struct sstable_manager_service {
+    db::nop_large_data_handler large_data_handler;
+    db::config dbcfg;
+    gms::feature_service feature_service;
+    cache_tracker tracker;
+    sstables::directory_semaphore dir_sem;
+    sstables::sstables_manager sst_man;
+
+    explicit sstable_manager_service()
+        : feature_service(gms::feature_config_from_db_config(dbcfg))
+        , dir_sem(1)
+        , sst_man(large_data_handler, dbcfg, feature_service, tracker, memory::stats().total_memory(), dir_sem) {
+    }
+
+    future<> stop() {
+        return sst_man.close();
+    }
+};
+
+mutation_opt read_schema_table_mutation(sharded<sstable_manager_service>& sst_man, std::filesystem::path schema_table_data_path,
+        std::function<schema_ptr()> schema_factory, reader_permit permit, std::string_view keyspace, std::vector<std::string_view> ck_strings) {
+
+    sharded<sstables::sstable_directory> sst_dirs;
+    sst_dirs.start(
+        sharded_parameter([&sst_man] { return std::ref(sst_man.local().sst_man); }),
+        sharded_parameter([&schema_factory] { return schema_factory(); }),
+        schema_table_data_path,
+        sharded_parameter([] { return default_priority_class(); }),
+        sharded_parameter([] { return default_io_error_handler_gen(); })).get();
+    auto stop_sst_dirs = deferred_stop(sst_dirs);
+
+    auto sstable_open_infos = sst_dirs.map_reduce0(
+            [] (sstables::sstable_directory& sst_dir) -> future<std::vector<sstables::foreign_sstable_open_info>> {
+               co_await sst_dir.process_sstable_dir(sstables::sstable_directory::process_flags{ .sort_sstables_according_to_owner = false });
+               const auto& unsorted_ssts = sst_dir.get_unsorted_sstables();
+               std::vector<sstables::foreign_sstable_open_info> open_infos;
+               open_infos.reserve(unsorted_ssts.size());
+               for (auto& sst : unsorted_ssts) {
+                   open_infos.push_back(co_await sst->get_open_info());
+               }
+               co_return open_infos;
+            },
+            std::vector<sstables::foreign_sstable_open_info>{},
+            [] (std::vector<sstables::foreign_sstable_open_info> a, std::vector<sstables::foreign_sstable_open_info> b) {
+                std::move(b.begin(), b.end(), std::back_inserter(a));
+                return a;
+            }).get();
+
+    auto schema_table_schema = schema_factory();
+
+    if (sstable_open_infos.empty()) {
+        return {};
+    }
+
+    std::vector<sstables::shared_sstable> sstables;
+    sstables.reserve(sstable_open_infos.size());
+    for (auto& open_info : sstable_open_infos) {
+        sstables.push_back(sst_dirs.local().load_foreign_sstable(open_info).get());
+    }
```

Comment:
get0 ?

Reply:
For a while now (a few years actually), `get0()` is identical to `get()`. In fact I think at some point we will deprecate `get0()`, its only there for backwards compatibility.